src/components/button/button.style.js

```diff
@@ -0,0 +1,131 @@
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import BaseTheme from '../../style/themes/base';
+import buttonTypes from './button-types.style';
+import buttonSizes from './button-sizes.style';
+import classicConfig from './button-classic-config.style';
+import OptionsHelper from '../../utils/helpers/options-helper';
+
+const StyledButton = styled.button`
+  align-items: center;
+  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
+  display: inline-flex;
+  flex-direction: column;
+  flex-flow: wrap;
+  justify-content: center;
+  vertical-align: middle;
+
+  ${({ theme }) => theme.name !== 'classic' && stylingForType}
+  ${({ theme, size }) => theme.name !== 'classic' && buttonSizes(theme)[size]}
+
+  ${({ theme, renderAs }) => theme.name === 'classic' && classicRenderAs(renderAs) && stylingForClassic}
```

Comment:
maybe it could be done in one method?
```
${({ theme, renderAs, size }) => ...
```

Reply:
Yeah I've tidied this up a bitand separated it into a function that calls the correct style based on the theme context etc